lib/srv/db/sqlserver/connect.go

```diff
@@ -38,7 +42,36 @@ type Connector interface {
 }
 
 type connector struct {
-	Auth common.Auth
+	// Auth is the database auth client
+	DBAuth common.Auth
+	// AuthClient is the teleport client
+	AuthClient windows.AuthInterface
+	// DataDir is the Teleport data directory
+	DataDir string
+
+	kinitCommandGenerator kinit.CommandGenerator
+	caFunc                func(ctx context.Context, clusterName string) ([]byte, error)
+}
+
+var errBadKerberosConfig = errors.New("configuration must have either keytab or kdc_host_name and ldap_cert")
+
+func (c *connector) getKerberosClient(ctx context.Context, sessionCtx *common.Session) (*client.Client, error) {
+	var kt *client.Client
+	var err error
+	if sessionCtx.Database.GetAD().KeytabFile != "" {
+		kt, err = c.keytabClient(sessionCtx)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+	} else if sessionCtx.Database.GetAD().KDCHostName != "" && sessionCtx.Database.GetAD().LDAPCert != "" {
+		kt, err = c.kinitClient(ctx, sessionCtx, c.AuthClient, c.DataDir)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+	} else {
+		return nil, trace.Wrap(errBadKerberosConfig)
+	}
+	return kt, nil
```

Comment:
nit: just a suggestion: 
```suggestion
func (c *connector) getKerberosClient(ctx context.Context, sessionCtx *common.Session) (*client.Client, error) {
       switch {
	case sessionCtx.Database.GetAD().KeytabFile != "":
		kt, err = c.keytabClient(sessionCtx)
		if err != nil {
			return nil, trace.Wrap(err)
		}
		return kt, nil
	case sessionCtx.Database.GetAD().KDCHostName != "" && sessionCtx.Database.GetAD().LDAPCert != "":
		kt, err = c.kinitClient(ctx, sessionCtx, c.AuthClient, c.DataDir)
		if err != nil {
			return nil, trace.Wrap(err)
		}
		return kt, nil
		
	}
	return nil, trace.Wrap(errBadKerberosConfig)
```